langroid/agent/chat_agent.py

```diff
@@ -145,10 +172,22 @@ def __init__(
         self.system_tool_format_instructions: str = ""
 
         self.llm_functions_map: Dict[str, LLMFunctionSpec] = {}
+        self.llm_functions_class_map: Dict[str, type[ToolMessage]] = {}
         self.llm_functions_handled: Set[str] = set()
         self.llm_functions_usable: Set[str] = set()
         self.llm_function_force: Optional[Dict[str, str]] = None
 
+        self.output_format: Optional[type[ToolMessage | BaseModel]] = None
+
+        self.saved_requests_and_tool_setings = self._requests_and_tool_settings()
+        if config.output_format is not None:
+            self.set_output_format(config.output_format)
+        # OpenAI does not support all strict schemas. We disable for
```

Comment:
Comment may need to be fixed? 
"controls whether to disable strict schemas for this agent if strict mode causes exception"